<launch>
  <arg name="first_tb3_prefix" default="tb3_0"/>
  <arg name="first_tb3_model"  default="burger_5"/>
  <arg name="first_tb3_x" 	default=" 0.0"/>
  <arg name="first_tb3_y" 	default=" 0.0"/>
  <arg name="first_tb3_z" 	default=" 0.0"/>
  <arg name="first_tb3_yaw"    default=" 0.0"/>
  <arg name="first_tb3_id"     default=" 5"/>
  
  <arg name="landmark_1_x"    default=" 1.0"/>
  <arg name="landmark_1_y"    default=" 0.0"/>
  <arg name="landmark_1_yaw"  default=" 1.57079632679"/>
  <arg name="landmark_1_id"   default="0"/>
  <arg name="landmark_1_name" default="aruco_marker_0"/>

  <arg name="dim_state"  default="6"/>
  <arg name="marker_length"  default="0.1"/>
  <arg name="aruco_cal_file"  default="calibrationSave_gazebo.p"/>
  <arg name="data_skip"  default="12"/>
  
  <!-- Launch gazebo world -->
  <arg name="debug" default="false" />
  <arg name="gui" default="true" />
  <arg name="pause" default="false" />
  <arg name="world" default="$(find dse_simulation)/world/aruco_test.world" />
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world)" />
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg pause)" />
    <arg name="use_sim_time" value="true" />
  </include>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find dse_simulation)/rviz/3_agents_4_tags_dse_odom_config.rviz" />

    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find dse_turtlebot_descriptions)/urdf/turtlebot3_$(arg first_tb3_model).urdf.xacro" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <!--<param name="tf_prefix" value="$(arg first_tb3_prefix)" />-->
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg first_tb3_prefix) -x $(arg first_tb3_x) -y $(arg first_tb3_y) -z $(arg first_tb3_z) -Y $(arg first_tb3_yaw) -param robot_description" />
  
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg first_tb3_prefix)_tf_broadcaster"  args="0 0 0 0 0 0 /world /odom 100"/>
  
  <!--<group ns = "$(arg first_tb3_prefix)">
    <node pkg="dse_simulation" type="aruco_validation.py" name="aruco_pose_estimation_node" output="screen">
      <param name="prefix" value="$(arg first_tb3_prefix)" />
      <param name="marker_length" type="double" value="$(arg marker_length)" />
      <param name="calibration_file" value="$(arg aruco_cal_file)" />
      <param name="data_skip" value="$(arg data_skip)" />
    </node>
  </group>-->
</launch>
